### DISCUSSION TITLE
Conan 2.x lockfile support

### DISCUSSION THREAD

**Discussion by dus7eh** [aquasecurity/trivy#6065] [Ideas]
### Description

Add support for Conan 2.x for c++.

The format of `conan.lock` file has changed significantly in Conan 2.
Announcement: https://docs.conan.io/2/whatsnew.html#new-lockfiles
Lockfiles: https://docs.conan.io/2/tutorial/consuming_packages/intro_to_versioning.html#tutorial-consuming-packages-versioning-lockfiles

In projects with Conan 2 manager trivy fails to generate proper scanner reports.

### Target

Filesystem

### Scanner

None

**Comment by DmitriyLewen** [comment]
Hello @dus7eh 
Thanks for information!

Created #6066 for this task.

Regards, Dmitriy
